app/containers/TracksContainer/index.js

```diff
@@ -1,5 +1,159 @@
+import TrackComponent from '@app/components/TrackComponent/index';
 
-export default function TracksContainer() {
-  return <div> TracksContainer page </div>;
+const { Search } = Input;
+
+const TitleCard = styled(Card)`
+  && {
+    margin: 10px 0;
+    max-width: ${props => props.maxwidth}px;
+    color: ${props => props.color}px;
+  }
+`;
+
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    max-width: ${props => props.maxwidth}px;
+    width: 100%;
+    margin: 0 auto;
+    padding: ${props => props.padding}px;
+  }
+`;
+
+export function TracksContainer({
+  maxwidth,
+  padding,
+  tracksError,
+  tracksData,
+  trackName,
+  intl,
+  dispatchTrackNames,
+  dispatchClearTracksData
+}) {
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loadedTracksData = get(tracksData, 'results', null) || tracksError;
+    if (loading && loadedTracksData) {
+      setLoading(false);
+    }
+  }, [tracksData]);
+
+  useEffect(() => {
+    if (trackName && !tracksData?.results?.length) {
+      dispatchTrackNames(trackName);
+      setLoading(true);
+    }
+  }, []);
+
+  const handleOnChange = trackname => {
+    if (!isEmpty(trackname)) {
+      dispatchTrackNames(trackname);
+      setLoading(true);
+    } else {
+      dispatchClearTracksData();
+    }
+  };
+
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  const renderTracksList = () => {
+    const loadedTrackSongs = get(tracksData, 'results', []);
+
+    return (
+      <If condition={!isEmpty(loadedTrackSongs) || loading}>
+        <TitleCard>
+          <Skeleton loading={loading} active>
+            <For
+              of={loadedTrackSongs}
+              ParentComponent={Container}
+              renderItem={(item, index) => <TrackComponent key={index} {...item} />}
+            />
+          </Skeleton>
+        </TitleCard>
+      </If>
+    );
+  };
+
+  return (
+    <Container maxwidth={maxwidth} padding={padding} data-testid="tracks-container">
+      <TitleCard title={intl.formatMessage({ id: 'itunes_title' })} maxwidth={maxwidth}>
+        <T id="track_search_default" paddingBottom={5} />
+        <Search
+          type="text"
+          data-testid="search-bar"
+          defaultValue={trackName}
+          onChange={e => debouncedHandleOnChange(e.target.value)}
+        />
+      </TitleCard>
+
+      {renderTracksList()}
+    </Container>
+  );
 }
+
+TracksContainer.propTypes = {
+  maxwidth: PropTypes.number,
+  padding: PropTypes.number,
+  trackName: PropTypes.string,
+  dispatchTrackNames: PropTypes.func,
+  dispatchClearTracksData: PropTypes.func,
+  intl: PropTypes.object,
+  tracksError: PropTypes.object,
+  tracksData: PropTypes.shape({
+    results: PropTypes.array
+  })
+};
+
+TracksContainer.defaultProps = {
+  maxwidth: 500,
+  padding: 20
+};
+
+const mapStateToProps = createStructuredSelector({
+  tracksContainer: selectTracksContainer(),
```

Comment:
Why is there tracksContainer that you are trying to select ?

Reply:
Fixed this.